src/lib/device/virtual_dev.cpp

```diff
@@ -123,6 +123,14 @@ void VirtualDev::add_chunk(cshared< Chunk >& chunk, bool is_fresh_chunk) {
     m_chunk_selector->add_chunk(chunk);
 }
 
+void VirtualDev::remove_chunk(cshared< Chunk >& chunk) {
+    std::unique_lock lg{m_mgmt_mutex};
+    auto iter = std::remove_if(m_all_chunks.begin(), m_all_chunks.end(), [chunk](auto c) { return c == chunk; });
+    m_all_chunks.erase(iter, m_all_chunks.end());
+
+    // m_chunk_selector->remove_chunk(chunk); // TODO
```

Comment:
won't there be problem if it is not removed from heap_chunk_selector, is there a reason it can't be done within this PR?

Reply:
Added remove_chunk() default function. HO can override it.